include/sparrow/buffer.hpp

```diff
@@ -90,6 +90,34 @@ namespace sparrow
     template <class T>
     bool operator==(const buffer<T>& lhs, const buffer<T>& rhs);
 
+    /*
+     * @class buffer_view
+     * @brief Object that references but does not own a piece of contiguous memory
+     */
+    template <class T>
+    class buffer_view : private impl::buffer_data<T>
+    {
+    public:
+
+        using base_type = impl::buffer_data<T>;
+        using value_type = typename base_type::value_type;
+        using pointer = typename base_type::pointer;
+        using size_type = typename base_type::size_type;
+
+        explicit buffer_view(buffer<T>& buffer);
+        buffer_view(pointer data, size_type size);
+
+        using base_type::empty;
```

Comment:
This is not covered by the tests.

Should it be?

Reply:
Yep it should, will update the PR.